sdk/communication/chat/src/main/java/com/azure/android/communication/chat/AzureCommunicationChatServiceAsyncClient.java

```diff
@@ -0,0 +1,490 @@
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatThreadsPages(final Callback<AsyncPagedDataCollection<ChatThreadInfo, Page<ChatThreadInfo>>> collectionCallback, Integer maxPageSize, OffsetDateTime startTime) {
+        this.serviceClient.listChatThreadsPagesAsync(collectionCallback, maxPageSize, startTime);
+    }
+
+    /**
+     * Updates a thread's properties.
+     * 
+     * @param chatThreadId The id of the thread to update.
+     * @param body Request payload for updating a chat thread.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void updateChatThread(String chatThreadId, UpdateChatThreadOptions body, final Callback<Void> callback) {
+        this.serviceClient.updateChatThread(chatThreadId, body, callback);
+    }
+
+    /**
+     * Gets a chat thread.
+     * 
+     * @param chatThreadId Thread id to get.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void getChatThread(String chatThreadId, final Callback<ChatThread> callback) {
+        this.serviceClient.getChatThread(chatThreadId, callback);
+    }
+
+    /**
+     * Deletes a thread.
+     * 
+     * @param chatThreadId Thread id to delete.
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void deleteChatThread(String chatThreadId, final Callback<Void> callback) {
+        this.serviceClient.deleteChatThread(chatThreadId, callback);
+    }
+
+    /**
+     * Get the next page of items.
+     * 
+     * @param nextLink null
+     * @param callback the Callback that receives the response.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    public void listChatReadReceiptsNext(String nextLink, final Callback<Page<ReadReceipt>> callback) {
```

Comment:
I think these should be private or protected, no?

Reply:
it could be, as long as page retriever can access with strict modifiers. Currently, it's following the visibility of autogen Java.

Reply:
We have to decide whether to expose these pairs for pagination: list and listNext. For now more is better IMHO.